wso2-synapse-unit-test-plugin/src/main/java/org/wso2/synapse/unittest/UnitTestCasesMojo.java

```diff
@@ -64,25 +87,40 @@ public void execute() throws MojoExecutionException {
 
     /**
      * Execution method of Mojo class.
+     *
+     * @throws IOException if error occurred while reading test files
      */
     private void testCaseRunner() throws IOException {
-
-        ArrayList<String> synapseTestCasePaths = getTestCasesFileNamesWithPaths(testCasesFilePath);
+        List<String> synapseTestCasePaths = getTestCasesFileNamesWithPaths(testCasesFilePath);
 
         getLog().info("Detect " + synapseTestCasePaths.size() + " Synapse test case files to execute\n");
+        getLog().info("");
+
+        //start the synapse engine with enable the unit test agent
+        if (!synapseTestCasePaths.isEmpty()) {
+            startTestingServer();
+        }
 
+        Map<String, String> testSummaryData = new HashMap<>();
         for (String synapseTestCaseFile : synapseTestCasePaths) {
 
             String responseFromUnitTestFramework = UnitTestClient.executeTests
-                    (synapseTestCaseFile, synapseServer.getHost(), synapseServer.getUnitTestPort());
+                    (synapseTestCaseFile, serverHost, serverPort);
 
-            if (responseFromUnitTestFramework != null) {
-                isAssertResults(responseFromUnitTestFramework, synapseTestCaseFile);
+            if (responseFromUnitTestFramework != null
+                    && !responseFromUnitTestFramework.equals(Constants.NO_TEST_CASES)) {
+                testSummaryData.put(synapseTestCaseFile, responseFromUnitTestFramework);
             } else {
-                throw new IOException("Test case parsing failed ");
+                getLog().info("No test cases found in " + synapseTestCaseFile + " unit test suite");
+                getLog().info("");
```

Comment:
what is this empty info log?

Reply:
To keep the empty line with [INFO] tag. Because if I used the line separator in the previous log line it prints empty line without [INFO] tag.